### PR TITLE
Update windows certificate store so that the client certificate is no longer needer in certificate store

### PR DESCRIPTION
**Pull request by mattdurham**
#### PR Description

Remove the need to have the client certificate  in the store, most of the changes are deletion of finding the client certificate.

#### Notes to the Reviewer

moved some code around for better visibility

#### PR Checklist

- [X] CHANGELOG updated
- [X] Documentation added
- [X] Tests updated

### CODE REVIEW

File: pkg/server/tls_certstore_windows_test.go [hunk 1]
```diff
@@ -154,64 +153,6 @@ func TestMatching2CertsGetMostRecent(t *testing.T) {
 	require.Equal(t, foundCert, shouldFind)
 }
 
-func TestRegularExpression(t *testing.T) {
```
Comment (marctc) [inline pkg/server/tls_certstore_windows_test.go:157]: Why is this code removed? I didn't see any removal of regex logic in `tls_certore_windows.go` file.

File: pkg/server/tls_certstore_windows.go [hunk 2]
```diff
@@ -7,16 +7,32 @@ import (
...
+	"github.com/go-kit/log/level"
 	"regexp"
 	"sort"
 	"sync"
 	"time"
+)
 
-	"github.com/go-kit/log"
-	"github.com/go-kit/log/level"
+// winCertStoreHandler handles the finding of certificates, validating them and injecting into the default tls pipeline
```
Comment (marctc) [inline pkg/server/tls_certstore_windows.go:19]: nit: capitalize tls

File: pkg/server/tls_certstore_windows_test.go (hunk 1 above)
Comment (mattdurham) [reply pkg/server/tls_certstore_windows_test.go:157]: Before it was using it to find the subject in a complex manner, now its a more simple match.

File: CHANGELOG.md [hunk 3]
```diff
@@ -40,6 +40,7 @@ Main (unreleased)
 - Relative symlinks for promtail now work as expected. (@RangerCD, @mukerjee)
 - Fix rate limiting implementation for the app agent receiver integration. (@domasx2)
 - Fix mongodb exporter so that it now collects all metrics. (@mattdurham)
+- Change windows certificate store so client certificate is no longer required in store (@mattdurham)
```
Comment (rfratto) [inline CHANGELOG.md:43]: Technically this is a breaking change, can we document it as one?

**Review by rfratto — APPROVED** [review]

File: pkg/server/tls_certstore_windows.go [hunk 4]
```diff
@@ -102,25 +111,55 @@ func (l *tlsListener) applyWindowsCertificateStore(c TLSConfig) error {
...
+// VerifyPeer is called by the TLS pipeline, and specified in tls.config, this is where any custom verification happens
+func (c *winCertStoreHandler) VerifyPeer(_ [][]byte, verifiedChains [][]*x509.Certificate) error {
+	opts := x509.VerifyOptions{}
+	clientCert := verifiedChains[0][0]
 
-type winCertStoreHandler struct {
-	cfg          WindowsCertificateFilter
-	subjectRegEx *regexp.Regexp
-	log          log.Logger
+	// Check for issuer
```
Comment (tpaschalis) [inline pkg/server/tls_certstore_windows.go:119]: This is the main body that changes the behavior, right?

Comment (mattdurham) [reply pkg/server/tls_certstore_windows.go:119]: Yes, the checking is now explicitly done in VerifyPeer

**Pull request merged**